src/components/input-time/input-time.tsx

```diff
@@ -0,0 +1,872 @@
+import {
+  Component,
+  Element,
+  h,
+  Host,
+  Prop,
+  VNode,
+  Event,
+  EventEmitter,
+  Watch,
+  State,
+  Listen,
+  Method
+} from "@stencil/core";
+import { Scale } from "../interfaces";
+import { numberKeys } from "../../utils/key";
+import { isValidNumber } from "../../utils/number";
+import { LabelableComponent, connectLabel, disconnectLabel } from "../../utils/label";
+import {
+  connectForm,
+  disconnectForm,
+  FormComponent,
+  HiddenFormInputSlot,
+  submitForm
+} from "../../utils/form";
+import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
+import {
+  formatTimePart,
+  formatTimeString,
+  MinuteOrSecond,
+  maxTenthForMinuteAndSecond,
+  TimePart,
+  getMeridiem,
+  HourCycle,
+  localizeTimeStringToParts,
+  parseTimeString,
+  localizeTimePart,
+  Meridiem,
+  getLocaleHourCycle,
+  getLocaleHourSuffix,
+  getLocaleMinuteSuffix,
+  getLocaleSecondSuffix,
+  getTimeParts
+} from "../../utils/time";
+import { CSS, TEXT } from "./resources";
+import { NumberingSystem } from "../../utils/locale";
+import {
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent
+} from "../../utils/loadable";
+
+function capitalize(str: string): string {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+@Component({
+  tag: "calcite-input-time",
+  styleUrl: "input-time.scss",
+  shadow: true
+})
+export class InputTime
+  implements LabelableComponent, FormComponent, InteractiveComponent, LoadableComponent
+{
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteInputTimeElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** The disabled state of the time input */
+  @Prop({ reflect: true }) disabled = false;
+
+  /** When true, the icon is flipped in RTL. */
+  @Prop({ reflect: true }) iconFlipRtl = false;
+
+  /**
+   * aria-label for the hour input
+   *
+   * @default "Hour"
+   */
+  @Prop() intlHour = TEXT.hour;
+
+  /**
+   * aria-label for the meridiem (am/pm) input
+   *
+   * @default "AM/PM"
+   */
+  @Prop() intlMeridiem = TEXT.meridiem;
+
+  /**
+   * aria-label for the minute input
+   *
+   * @default "Minute"
+   */
+  @Prop() intlMinute = TEXT.minute;
```

Comment:
any label's that needs to be translated should use `messages` & `messageOverrides` properties. Please ref component with `t9nComponent` interface.